src/lib/operators/jit_operator/operators/jit_write_reference.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "abstract_jittable_sink.hpp"
+
+namespace opossum {
+
+/* JitWriteReference must be the last operator in any chain of jit operators.
+ * It is responsible for
+ * 1) adding column definitions to the output table
+ * 2) appending the current row id to the current output chunk
+ * 3) creating a new output chunk and adding output chunks to the output table
+ */
+class JitWriteReference : public AbstractJittableSink {
```

Comment:
Maybe add a comment that this is called "JitWriteReference" because it writes one reference for each call of `_consume`. This is probably one of only few places where we don't use the plural.

Reply:
I changed `JitWriteReference` to `JitWriteReferences` to keep consistent with the naming convention of `JitReadTuples` and `JitWriteTuples`. `JitWriteTuples` also just writes a single tuple with multiple values in `_consume` but uses the plural of tuple.